numpydoc/numpydoc.py

```diff
@@ -206,6 +206,8 @@ def mangle_signature(app, what, name, obj, options, sig, retann):
     sig = doc['Signature'] or getattr(obj, '__text_signature__', None)
     if sig:
         sig = re.sub("^[^(]*", "", sig)
+        sig = re.sub(r"\$self,\s", "", sig)
```

Comment:
I think if the signature only has `$self` and no other arguments this may not be correct since it might look like `($self)` not `($self, )`.

Reply:
Hmm... I think you're right. 

Thinking about it more, maybe I should write a helper function that only us used on the `__text_signature__` field. I think essentially it would be something like... 

```
def clean_text_signature(__text_signature__):
    params = ___text_signature__.split(',')
    params.remove('$self')
    params.remove('/')
    if len(params) > 1:
        params = ' ,'.join(params)
    return '(' + params + ')'
```

Reply:
Sounds good, but probably remove $type, $module too?

Reply:
done